private/dependency_tree_parser.bzl

```diff
@@ -330,3 +331,39 @@ def _generate_imports(repository_ctx, dep_tree, explicit_artifacts, neverlink_ar
 parser = struct(
     generate_imports = _generate_imports,
 )
+
+def _simple_coord_matches_pattern(simple_coord, patterns):
+    for raw_pattern in patterns:
+        pattern_and_value=_parse_raw_pattern(raw_pattern)
+
+        pattern = pattern_and_value[0]
+        token = pattern_and_value[1]
+
+        if (
+                (pattern == "equals" and simple_coord == token)
+                or
+                (pattern == "contains" and token in simple_coord)
```

Comment:
Is simple_coord a list? If so, make it a dict or something so that the `in` operator is o(1) and not o(n).